src/Umbraco.Cloud.StorageProviders.AzureBlob/AzureBlobComposer.cs

```diff
@@ -1,35 +1,45 @@
 using System;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Umbraco.Cms.Core.Composing;
 using Umbraco.Cms.Core.DependencyInjection;
 using Umbraco.Cms.Web.Common.ApplicationBuilder;
-using Umbraco.StorageProviders.AzureBlob;
 
 namespace Umbraco.Cloud.StorageProviders.AzureBlob
 {
+    /// <summary>
+    /// Automatically configures Azure Blob Storage for use on Umbraco Cloud.
+    /// </summary>
+    /// <seealso cref="Umbraco.Cms.Core.Composing.IComposer" />
     public class AzureBlobComposer : IComposer
     {
+        /// <inheritdoc />
         public void Compose(IUmbracoBuilder builder)
         {
             if (builder == null) throw new ArgumentNullException(nameof(builder));
 
-            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("UMBRACO__CLOUD__ENVIRONMENT"))) return;
-
-            builder.AddAzureBlobMediaFileSystem();
-
             var configuration = new ConfigurationBuilder()
                 .AddEnvironmentVariables("Umbraco:Cloud:")
                 .Build();
 
-            builder.Services.Configure<AzureBlobFileSystemOptions>(AzureBlobFileSystemOptions.MediaFileSystemName,
-                options =>
-                {
-                    var section = configuration.GetSection("Storage:AzureBlob");
+            // Only configure when running on Umbraco Cloud with valid options
+            var isRunningOnCloud = configuration.GetValue<bool>("IsRunningOnCloud");
+            if (isRunningOnCloud == false) return;
 
-                    options.ConnectionString = $"BlobEndpoint={section["Endpoint"]};SharedAccessSignature={section["SharedAccessSignature"]}";
-                    options.ContainerName = section["ContainerName"];
-                });
+            // Get options and manually validate (no need to register them into DI)
+            var azureBlobOptions = configuration.GetSection("Storage:AzureBlob").Get<AzureBlobOptions>();
```

Comment:
Should we update this to `Storage:AzureBlob:Media`, so it's the same as the base package configuration and we have the possibility to setup other file systems (e.g. for Forms) in the future?
```suggestion
            var azureBlobOptions = configuration.GetSection("Storage:AzureBlob:Media").Get<AzureBlobOptions>();
```

Reply:
I we need to have a forms filesystem it'll use the same credentials (we already do this for Heartcore on v8), so there's no need to change this

Reply:
That makes sense, as both media and cache files are already stored in separate directories within the blob container 👍🏻